modules/cli/src/main/scala/scala/cli/commands/util/ScalacOptionsUtil.scala

```diff
@@ -0,0 +1,21 @@
+package scala.cli.commands.util
+
+import scala.build.options.{ScalacOpt, ShadowingSeq}
+import scala.cli.commands.ScalacOptions
+
+object ScalacOptionsUtil {
+  extension (opts: List[String]) {
+    def toShadowingSeq: ShadowingSeq[ScalacOpt] =
```

Comment:
This looks much better for me, I have one suggestion, we should name better this `extension`, because it can be used for any `List[String]`.
```suggestion
    def toScalacOptShadowingSeq: ShadowingSeq[ScalacOpt] =
```

Reply:
fixed